defaultTo.js

```diff
@@ -15,9 +20,15 @@
  *
  * defaultTo(undefined, 10)
  * // => 10
+ *
+ * defaultTo('', 'default', isEmpty)
+ * // => 'default'
+ *
+ * defaultTo(new Error(), 'default', isError)
+ * // => 'default'
  */
-function defaultTo(value, defaultValue) {
-  return (value == null || value !== value) ? defaultValue : value
+function defaultTo(value, defaultValue, resolveEmpty = overSome([isNaN, isNil])) {
```

Comment:
As the [first example in the PR description](https://github.com/lodash/lodash/pull/4267#issue-270557882) illustrates, the main note I have on this implementation is that `resolveEmpty` is not equivalent to `lodash.isEmpty` or indeed any single existing Lodash function, and this was the existing behaviour of `defaultTo` beforehand.

Altering this of course would be breaking, so the default value for `resolveEmpty` behaves identically. I believe `overSome([isNaN, isNil])` is the most succinct way to redefine the `(value == null || value !== value)` test by using tested Lodash functions.